src/main/java/edu/kis/powp/jobs2d/command/gui/ComplexCommandEditorUI.java

```diff
@@ -0,0 +1,131 @@
+package edu.kis.powp.jobs2d.command.gui;
+
+import edu.kis.powp.jobs2d.command.*;
+import edu.kis.powp.jobs2d.command.manager.DriverCommandManager;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+
+public class ComplexCommandEditorUI extends JFrame {
+    private final DriverCommandManager manager;
+    private ComplexCommandEditor editor;
+    private DefaultListModel<DriverCommand> listModel;
+    private JList<DriverCommand> commandList;
+
+    public ComplexCommandEditorUI(DriverCommandManager manager) {
+        this.manager = manager;
+
+        if (manager.getCurrentCommand() instanceof ICompoundCommand) {
+            this.editor = new ComplexCommandEditor((ICompoundCommand) manager.getCurrentCommand());
+        } else {
+            this.editor = new ComplexCommandEditor();
+        }
+
+        setupUI();
+    }
+
+    private void setupUI() {
```

Comment:
Too large method. Please clean-up (e.g., extract commands to separate methods or classes).